File: Behavioral/State/CSharp/TrafficLight.cs
// TrafficLight.cs

public class TrafficLight {
  private ITrafficLightState _state;

  public TrafficLight() {
    _state = new RedLight(); // Initial state
  }

  public void SetState(ITrafficLightState state) {
    _state = state;
  }

  public void Change() {
    _state.Handle(this);
  }
}

File: Behavioral/State/JavaScript/trafficLight.js
// trafficLight.js

const RedLight = require("./redLight");

class TrafficLight {
  constructor() {
    this.state = new RedLight(); // Initial state
  }

  setState(state) {
    this.state = state;
  }

  change() {
    this.state.handle(this);
  }
}

module.exports = TrafficLight;
